xcodeproj/internal/xcode_schemes_internal.bzl

```diff
@@ -0,0 +1,130 @@
+"""Internal API for creating and manipulating Xcode schemes"""
+
+load(":target_id.bzl", "get_id")
+
+def _scheme(
+        name,
+        build_action = None,
+        test_action = None,
+        launch_action = None):
+    """Returns a `struct` representing an Xcode scheme.
+
+    Args:
+        name: The user-visible name for the scheme as a `string`.
+        build_action: Optional. A `struct` as returned by
+            `xcode_schemes.build_action`.
+        test_action: Optional. A `struct` as returned by
+            `xcode_schemes.test_action`.
+        launch_action: Optional. A `struct` as returned by
+            `xcode_schemes.launch_action`.
+
+    Returns:
+        A `struct` representing an Xcode scheme.
+    """
+    return struct(
+        name = name,
+        build_action = build_action,
+        test_action = test_action,
+        launch_action = launch_action,
+    )
+
+def _build_action(targets):
+    """Constructs a build action for an Xcode scheme.
+
+    Args:
+        targets: A `sequence` of target labels as `string` values.
+
+    Return:
+        A `struct` representing a build action.
+    """
+    return struct(
+        targets = targets,
+    )
+
+def _test_action(targets):
+    """Constructs a test action for an Xcode scheme.
+
+    Args:
+        targets: A `sequence` of target labels as `string` values.
+
+    Return:
+        A `struct` representing a test action.
+    """
+    return struct(
+        targets = targets,
+    )
+
+def _launch_action(
+        target,
+        args = None,
+        env = None,
+        working_directory = None):
+    """Constructs a launch action for an Xcode scheme.
+
+    Args:
+        target: A target label as a `string` value.
+        args: Optional. A `list` of `string` arguments that should be passed to
+            the target when executed.
+        env: Optional. A `dict` of `string` values that will be set as
+            environment variables when the target is executed.
+        working_directory: Optional. A `string` that will be set as the custom
+            working directory in the Xcode scheme's launch action.
+
+    Return:
+        A `struct` representing a launch action.
+    """
+    return struct(
+        target = target,
+        args = args,
+        env = env,
+        working_directory = working_directory,
+    )
+
+def _replace_labels_with_target_ids(scheme, configuration):
```

Comment:
I don't think this will work as you expect. The configuration on labels can be different. There might be multiple targets for the same configuration. The deps of a target might have a different configuration than their parent.

I believe we need to defer this to `generator`. There we can traverse the deps of a top level targets mentioned (or all if none are mentioned) and find the `TargetID` that best matches a given label.

Reply:
OK. I have a couple of questions.
1. For my edification, which configuration will be resolved [here](https://github.com/buildbuddy-io/rules_xcodeproj/blob/main/xcodeproj/internal/xcodeproj.bzl#L288)? Is this host configuration?
2. Is there a rule-of-thumb for when label-configuration should be resolved in Starlark code vs the generator? The JSON that the Starlark code sends to the generator only seems to send `TargetID` strings.

Reply:
I agree that we are going to iterate to pick the best one. I misunderstood where we needed to generate the list of all possible combinations. 

I will remove the label to `TargetID` code. Also, I will figure out why we are not seeing generator fixtures being updated.